cmd/user-cluster-controller-manager/main.go

```diff
@@ -265,6 +265,7 @@ func main() {
 			MLAGatewayURL:                runOp.mlaGatewayURL,
 			PrometheusScrapeConfigPrefix: runOp.prometheusScrapeConfigPrefix,
 		},
+		runOp.clusterName,
```

Comment:
The `-cluster-name` flag is provided to the UCCM only if the cluster is being CCM/CSI migrated. Please see this: https://github.com/kubermatic/kubermatic/blob/880595020ad5c5853f7cf416e53a548bfbcc162f/pkg/resources/usercluster/deployment.go#L179-L182

You should:

1. Change that part to unconditionally provide the `-cluster-name` flag
2. Change the flag validation in the UCCM to require the flag to be provided unconditionally: https://github.com/kubermatic/kubermatic/blob/1023c026298f124a5e70ebbeba4d1070ab347b44/cmd/user-cluster-controller-manager/main.go#L149-L151

Reply:
Thanks, fixed